# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../openoffice-org/patches/gcc43.diff
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- UnoControls/source/controls/progressbar.cxx
+++ UnoControls/source/controls/progressbar.cxx
@@ -69,6 +69,7 @@
 #endif
 
 #include <math.h>
+#include <limits.h>
 
 //____________________________________________________________________________________________________________
 //	includes of my project
--- bridges/source/cpp_uno/gcc3_linux_powerpc/cpp2uno.cxx
+++ bridges/source/cpp_uno/gcc3_linux_powerpc/cpp2uno.cxx
@@ -36,6 +36,8 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_bridges.hxx"
 
+#include <string.h>
+
 #include <com/sun/star/uno/genfunc.hxx>
 #include <uno/data.h>
 #include <typelib/typedescription.hxx>
--- bridges/source/cpp_uno/gcc3_linux_powerpc/except.cxx
+++ bridges/source/cpp_uno/gcc3_linux_powerpc/except.cxx
@@ -49,7 +49,7 @@
 #include <com/sun/star/uno/genfunc.hxx>
 #include <typelib/typedescription.hxx>
 #include <uno/any2.h>
-
+#include <string.h>
 #include "share.hxx"
 
 
--- bridges/source/cpp_uno/gcc3_linux_powerpc64/cpp2uno.cxx
+++ bridges/source/cpp_uno/gcc3_linux_powerpc64/cpp2uno.cxx
@@ -44,7 +44,7 @@
 #include "bridges/cpp_uno/shared/cppinterfaceproxy.hxx"
 #include "bridges/cpp_uno/shared/types.hxx"
 #include "bridges/cpp_uno/shared/vtablefactory.hxx"
-
+#include <string.h>
 #include "share.hxx"
 #include <stdio.h>
 
--- bridges/source/cpp_uno/gcc3_linux_powerpc64/except.cxx
+++ bridges/source/cpp_uno/gcc3_linux_powerpc64/except.cxx
@@ -49,7 +49,7 @@
 #include <com/sun/star/uno/genfunc.hxx>
 #include <typelib/typedescription.hxx>
 #include <uno/any2.h>
-
+#include <string.h>
 #include "share.hxx"
 
 
--- bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx
+++ bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx
@@ -37,6 +37,7 @@
 #include "precompiled_bridges.hxx"
 
 #include <stdio.h>
+#include <string.h>
 #include <dlfcn.h>
 #include <cxxabi.h>
 #include <hash_map>
--- bridges/source/remote/context/context.cxx
+++ bridges/source/remote/context/context.cxx
@@ -36,6 +36,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_bridges.hxx"
 #include <stdio.h>
+#include <string.h>
 #include <list>
 #include <hash_map>
 #include <utility>
--- bridges/source/remote/urp/urp_marshal.hxx
+++ bridges/source/remote/urp/urp_marshal.hxx
@@ -54,6 +54,8 @@
 #include "urp_marshal_decl.hxx"
 #endif
 
+#include <string.h>
+
 struct remote_Interface;
 
 namespace bridges_urp
--- connectivity/source/commontools/sqlerror.cxx
+++ connectivity/source/commontools/sqlerror.cxx
@@ -46,6 +46,8 @@
 #include <cppuhelper/exc_hlp.hxx>
 #include <rtl/ustrbuf.hxx>
 
+#include <string.h>
+
 //........................................................................
 namespace connectivity
 {
--- connectivity/source/drivers/adabas/BConnection.cxx
+++ connectivity/source/drivers/adabas/BConnection.cxx
@@ -76,6 +76,8 @@
 #include <connectivity/sqlparse.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity::adabas;
 using namespace connectivity;
 using namespace ::com::sun::star::uno;
--- connectivity/source/drivers/evoab2/NDatabaseMetaData.cxx
+++ connectivity/source/drivers/evoab2/NDatabaseMetaData.cxx
@@ -63,6 +63,7 @@
 #include <com/sun/star/sdbc/ColumnSearch.hpp>
 
 #include <vector>
+#include <string.h>
 
 #ifndef _CONNECTIVITY_EVOAB_EVOLUTION_API_HXX_
 #include "EApi.h"
--- connectivity/source/drivers/hsqldb/HStorageAccess.cxx
+++ connectivity/source/drivers/hsqldb/HStorageAccess.cxx
@@ -60,6 +60,7 @@
 #include "diagnose_ex.h"
 #endif
 
+#include <string.h>
 
 using namespace ::com::sun::star::container;
 using namespace ::com::sun::star::uno;
--- connectivity/source/drivers/jdbc/Blob.cxx
+++ connectivity/source/drivers/jdbc/Blob.cxx
@@ -52,6 +52,8 @@
 #include <connectivity/dbexception.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity;
 //**************************************************************
 //************ Class: java.sql.Blob
--- connectivity/source/drivers/jdbc/CallableStatement.cxx
+++ connectivity/source/drivers/jdbc/CallableStatement.cxx
@@ -64,6 +64,8 @@
 #include <comphelper/sequence.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::beans;
--- connectivity/source/drivers/jdbc/InputStream.cxx
+++ connectivity/source/drivers/jdbc/InputStream.cxx
@@ -42,6 +42,8 @@
 #include "java/tools.hxx"
 #endif
 
+#include <string.h>
+
 using namespace connectivity;
 //**************************************************************
 //************ Class: java.io.InputStream
--- connectivity/source/drivers/jdbc/JStatement.cxx
+++ connectivity/source/drivers/jdbc/JStatement.cxx
@@ -74,6 +74,7 @@
 #include "resource/jdbc_log.hrc"
 
 #include <algorithm>
+#include <string.h>
 
 using namespace ::comphelper;
 using namespace connectivity;
--- connectivity/source/drivers/jdbc/PreparedStatement.cxx
+++ connectivity/source/drivers/jdbc/PreparedStatement.cxx
@@ -68,6 +68,8 @@
 #endif
 #include "resource/jdbc_log.hrc"
 
+#include <string.h>
+
 using namespace connectivity;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::beans;
--- connectivity/source/drivers/jdbc/Reader.cxx
+++ connectivity/source/drivers/jdbc/Reader.cxx
@@ -42,6 +42,7 @@
 #ifndef _INC_MEMORY
 //#include <memory.h>
 #endif
+#include <string.h>
 using namespace connectivity;
 //**************************************************************
 //************ Class: java.io.Reader
--- connectivity/source/drivers/jdbc/ResultSet.cxx
+++ connectivity/source/drivers/jdbc/ResultSet.cxx
@@ -97,6 +97,8 @@
 #include "connectivity/dbexception.hxx"
 #endif
 
+#include <string.h>
+
 using namespace ::comphelper;
 
 using namespace connectivity;
--- connectivity/source/drivers/odbc/OConnection.cxx
+++ connectivity/source/drivers/odbc/OConnection.cxx
@@ -81,6 +81,8 @@
 #include <connectivity/dbexception.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity::odbc;
 using namespace connectivity;
 using namespace dbtools;
--- connectivity/source/drivers/odbc/OPreparedStatement.cxx
+++ connectivity/source/drivers/odbc/OPreparedStatement.cxx
@@ -37,6 +37,7 @@
 #include "precompiled_connectivity.hxx"
 
 #include <stdio.h>
+#include <string.h>
 
 #ifndef _OSL_DIAGNOSE_H_
 #include <osl/diagnose.h>
--- connectivity/source/parse/sqlbison.y
+++ connectivity/source/parse/sqlbison.y
@@ -17,6 +17,7 @@
 //--------------------------------------------------------------------------
 
 #include <vector>
+#include <string.h>
 
 #ifndef _CONNECTIVITY_SQLNODE_HXX
 #include <connectivity/sqlnode.hxx>
--- connectivity/source/parse/sqlnode.cxx
+++ connectivity/source/parse/sqlnode.cxx
@@ -146,6 +146,7 @@
 #ifndef TOOLS_DIAGNOSE_EX_H
 #include <tools/diagnose_ex.h>
 #endif
+#include <string.h>
 
 
 using namespace ::com::sun::star::sdbc;
--- dbaccess/source/core/api/KeySet.cxx
+++ dbaccess/source/core/api/KeySet.cxx
@@ -116,6 +116,7 @@
 #ifndef _TOOLS_DEBUG_HXX
 #include <tools/debug.hxx>
 #endif
+#include <string.h>
 
 using namespace dbaccess;
 using namespace ::connectivity;
--- desktop/source/so_comp/services.cxx
+++ desktop/source/so_comp/services.cxx
@@ -61,6 +61,7 @@
 #include "oemjob.hxx"
 #include "evaluation.hxx"
 
+#include <string.h>
 
 using namespace rtl;
 using namespace ::com::sun::star::uno;
--- filter/source/flash/swffilter.cxx
+++ filter/source/flash/swffilter.cxx
@@ -88,6 +88,7 @@
 
 //#include <stdlib.h>
 //#include <windows.h>
+#include <string.h>
 
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::frame;
--- filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx
+++ filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx
@@ -35,7 +35,7 @@
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_filter.hxx"
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <ctype.h>
 #include <stdio.h>
--- filter/source/xmlfilterdetect/filterdetect.cxx
+++ filter/source/xmlfilterdetect/filterdetect.cxx
@@ -35,7 +35,7 @@
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_filter.hxx"
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <ctype.h>
 #include <stdio.h>
--- forms/source/component/ComboBox.cxx
+++ forms/source/component/ComboBox.cxx
@@ -68,6 +68,8 @@
 #include <tools/diagnose_ex.h>
 #include <unotools/sharedunocomponent.hxx>
 
+#include <limits.h>
+
 using namespace dbtools;
 
 //.........................................................................
--- framework/source/services/substitutepathvars.cxx
+++ framework/source/services/substitutepathvars.cxx
@@ -130,6 +130,8 @@
 
 #include <comphelper/configurationhelper.hxx>
 
+#include <string.h>
+
 //_________________________________________________________________________________________________________________
 //	Defines
 //_________________________________________________________________________________________________________________
--- hwpfilter/source/formula.cpp
+++ hwpfilter/source/formula.cpp
@@ -39,7 +39,7 @@
 #include "nodes.h"
 #include "mapping.h"
 #include "hwpeq.h"
-#include <iostream.h>
+#include <iostream>
 
 extern LinkedList<Node> nodelist;
 
--- i18npool/source/defaultnumberingprovider/defaultnumberingprovider.cxx
+++ i18npool/source/defaultnumberingprovider/defaultnumberingprovider.cxx
@@ -51,6 +51,7 @@
 #include <localedata.hxx>
 #include <nativenumbersupplier.hxx>
 #include <stdio.h>
+#include <string.h>
 
 #if OSL_DEBUG_LEVEL == 0 && !defined(NDEBUG)
 #define NDEBUG
--- i18npool/source/localedata/saxparser.cxx
+++ i18npool/source/localedata/saxparser.cxx
@@ -37,6 +37,7 @@
 #include "precompiled_i18npool.hxx"
 
 #include <stdio.h>
+#include <string.h>
 #include <stack>
 
 #include "sal/main.h"
--- i18npool/source/search/textsearch.cxx
+++ i18npool/source/search/textsearch.cxx
@@ -85,6 +85,8 @@
 #pragma warning( disable: 4786 )
 #endif
 
+#include <string.h>
+
 using namespace ::com::sun::star::util;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::lang;
--- icc/SampleICC-1.3.2-bendian.patch
+++ icc/SampleICC-1.3.2-bendian.patch
@@ -5081,3 +5081,53 @@
   	IccProfLibTest \
   	IccStripUnknownTags \
   	IccV4ToMPE
+*** misc/SampleICC-1.3.2/Contrib/ICC_utils/CLUT.cpp	2007-08-20 22:04:53.000000000 +0200
+--- misc/build/SampleICC-1.3.2/Contrib/ICC_utils/CLUT.cpp	2008-01-31 13:53:16.000000000 +0100
+***************
+*** 77,82 ****
+--- 77,84 ----
+  //
+  //////////////////////////////////////////////////////////////////////
+  
++ #include <stdlib.h>
++ #include <string.h>
+  #include <iostream>
+  #include <fstream>
+  #include <sstream>
+*** misc/SampleICC-1.3.2/Contrib/ICC_utils/ICC_tool_exception.h	2007-08-20 22:04:54.000000000 +0200
+--- misc/build/SampleICC-1.3.2/Contrib/ICC_utils/ICC_tool_exception.h	2008-01-31 13:54:51.000000000 +0100
+***************
+*** 83,88 ****
+--- 83,89 ----
+  #ifndef __DEFINED_ICC_TOOL_EXCEPTION_H__
+  #define __DEFINED_ICC_TOOL_EXCEPTION_H__
+  
++ #include <string.h>
+  #include <exception>
+  #include <string>
+  
+*** misc/SampleICC-1.3.2/IccProfLib/IccCmm.cpp	2007-08-20 22:05:00.000000000 +0200
+--- misc/build/SampleICC-1.3.2/IccProfLib/IccCmm.cpp	2008-01-31 13:49:54.000000000 +0100
+***************
+*** 79,84 ****
+--- 79,86 ----
+  #pragma warning( disable: 4786) //disable warning in <list.h>
+  #endif
+  
++ #include <stdlib.h>
++ #include <string.h>
+  #include "IccCmm.h"
+  #include "IccTag.h"
+  #include "IccIO.h"
+*** misc/SampleICC-1.3.2/Contrib/ICC_utils/Vetters.h	2007-08-20 22:04:53.000000000 +0200
+--- misc/build/SampleICC-1.3.2/Contrib/ICC_utils/Vetters.h	2008-01-31 14:05:58.000000000 +0100
+***************
+*** 83,88 ****
+--- 83,89 ----
+  #include <string>
+  #include <sys/types.h>
+  #include <sys/stat.h>
++ #include <stdlib.h>
+  
+  #ifdef WIN32
+  typedef unsigned short mode_t;
--- icc/SampleICC-1.3.2.patch
+++ icc/SampleICC-1.3.2.patch
@@ -5034,3 +5034,53 @@
   	IccProfLibTest \
   	IccStripUnknownTags \
   	IccV4ToMPE
+*** misc/SampleICC-1.3.2/Contrib/ICC_utils/CLUT.cpp	2007-08-20 22:04:53.000000000 +0200
+--- misc/build/SampleICC-1.3.2/Contrib/ICC_utils/CLUT.cpp	2008-01-31 13:53:16.000000000 +0100
+***************
+*** 77,82 ****
+--- 77,84 ----
+  //
+  //////////////////////////////////////////////////////////////////////
+  
++ #include <stdlib.h>
++ #include <string.h>
+  #include <iostream>
+  #include <fstream>
+  #include <sstream>
+*** misc/SampleICC-1.3.2/Contrib/ICC_utils/ICC_tool_exception.h	2007-08-20 22:04:54.000000000 +0200
+--- misc/build/SampleICC-1.3.2/Contrib/ICC_utils/ICC_tool_exception.h	2008-01-31 13:54:51.000000000 +0100
+***************
+*** 83,88 ****
+--- 83,89 ----
+  #ifndef __DEFINED_ICC_TOOL_EXCEPTION_H__
+  #define __DEFINED_ICC_TOOL_EXCEPTION_H__
+  
++ #include <string.h>
+  #include <exception>
+  #include <string>
+  
+*** misc/SampleICC-1.3.2/IccProfLib/IccCmm.cpp	2007-08-20 22:05:00.000000000 +0200
+--- misc/build/SampleICC-1.3.2/IccProfLib/IccCmm.cpp	2008-01-31 13:49:54.000000000 +0100
+***************
+*** 79,84 ****
+--- 79,86 ----
+  #pragma warning( disable: 4786) //disable warning in <list.h>
+  #endif
+  
++ #include <stdlib.h>
++ #include <string.h>
+  #include "IccCmm.h"
+  #include "IccTag.h"
+  #include "IccIO.h"
+*** misc/SampleICC-1.3.2/Contrib/ICC_utils/Vetters.h	2007-08-20 22:04:53.000000000 +0200
+--- misc/build/SampleICC-1.3.2/Contrib/ICC_utils/Vetters.h	2008-01-31 14:05:58.000000000 +0100
+***************
+*** 83,88 ****
+--- 83,89 ----
+  #include <string>
+  #include <sys/types.h>
+  #include <sys/stat.h>
++ #include <stdlib.h>
+  
+  #ifdef WIN32
+  typedef unsigned short mode_t;
--- idlc/source/idlccompile.cxx
+++ idlc/source/idlccompile.cxx
@@ -71,6 +71,8 @@
 #endif
 #endif
 
+#include <string.h>
+
 using namespace ::rtl;
 using namespace ::osl;
 
--- idlc/source/idlcproduce.cxx
+++ idlc/source/idlcproduce.cxx
@@ -63,6 +63,8 @@
 #include <errno.h>
 #endif
 
+#include <string.h>
+
 using namespace ::rtl;
 using namespace ::osl;
 
--- jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
+++ jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx
@@ -38,6 +38,7 @@
 #if OSL_DEBUG_LEVEL > 0
 #include <stdio.h>
 #endif
+#include <string.h>
 
 #include "boost/scoped_array.hpp"
 #include "osl/diagnose.h"
--- jvmfwk/plugins/sunmajor/pluginlib/util.cxx
+++ jvmfwk/plugins/sunmajor/pluginlib/util.cxx
@@ -61,6 +61,7 @@
 #pragma warning(pop)
 #endif
 #endif
+#include <string.h>
 
 #include "sunjre.hxx"
 #include "vendorlist.hxx"
--- jvmfwk/source/elements.cxx
+++ jvmfwk/source/elements.cxx
@@ -50,6 +50,7 @@
 #include "libxml/xpathInternals.h"
 #include "rtl/bootstrap.hxx"
 #include "boost/optional.hpp"
+#include <string.h>
 // #define NS_JAVA_FRAMEWORK "http://openoffice.org/2004/java/framework/1.0"
 // #define NS_SCHEMA_INSTANCE "http://www.w3.org/2001/XMLSchema-instance"
 
--- jvmfwk/source/fwkutil.cxx
+++ jvmfwk/source/fwkutil.cxx
@@ -47,6 +47,7 @@
 #endif
 
 #include <string>
+#include <string.h>
 #include "osl/mutex.hxx"
 #include "osl/module.hxx"
 #include "osl/thread.hxx"
--- lingucomponent/source/languageguessing/guess.cxx
+++ lingucomponent/source/languageguessing/guess.cxx
@@ -37,6 +37,7 @@
 #include "precompiled_lingucomponent.hxx"
 
 #include <iostream>
+#include <string.h>
 
 #include <libtextcat/textcat.h>
 #include <altstrfunc.hxx>
--- package/source/zipapi/MemoryByteGrabber.hxx
+++ package/source/zipapi/MemoryByteGrabber.hxx
@@ -41,6 +41,7 @@
 #ifndef _COM_SUN_STAR_IO_XSEEKABLE_HPP_
 #include <com/sun/star/io/XSeekable.hpp>
 #endif
+#include <string.h>
 
 class MemoryByteGrabber 
 {
--- package/source/zipapi/XUnbufferedStream.cxx
+++ package/source/zipapi/XUnbufferedStream.cxx
@@ -60,6 +60,7 @@
 #include <EncryptedDataHeader.hxx>
 #endif
 #include <algorithm>
+#include <string.h>
 
 #if 0
 // for debugging purposes here
--- psprint/source/helper/helper.cxx
+++ psprint/source/helper/helper.cxx
@@ -39,6 +39,7 @@
 #include <cstring>
 #include <sys/stat.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <psprint/helper.hxx>
 #include <tools/string.hxx>
--- rdbmaker/source/rdbmaker/rdboptions.cxx
+++ rdbmaker/source/rdbmaker/rdboptions.cxx
@@ -33,6 +33,7 @@
  *
  ************************************************************************/
 #include 	<stdio.h>
+#include	<string.h>
 
 #include	"rdboptions.hxx"
 
--- registry/tools/checksingleton.cxx
+++ registry/tools/checksingleton.cxx
@@ -36,6 +36,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_registry.hxx"
 #include <stdio.h>
+#include <string.h>
 
 #include <set>
 
--- sax/source/expatwrap/sax_expat.cxx
+++ sax/source/expatwrap/sax_expat.cxx
@@ -33,6 +33,7 @@
  *
  ************************************************************************/
 #include <stdlib.h>
+#include <string.h>
 #include <sal/alloca.h>
 #include <vector>
 
--- sdext/source/minimizer/pppoptimizertoken.cxx
+++ sdext/source/minimizer/pppoptimizertoken.cxx
@@ -40,6 +40,7 @@
 #include <osl/mutex.hxx>
 #endif
 #include <hash_map>
+#include <string.h>
 
 struct TCheck
 {
--- shell/source/backends/gconfbe/gconflayer.cxx
+++ shell/source/backends/gconfbe/gconflayer.cxx
@@ -72,6 +72,8 @@
 #include <com/sun/star/uno/Sequence.hxx>
 #endif
 
+#include <string.h>
+
 using namespace rtl;
 
 //==============================================================================
--- shell/source/unix/sysshell/recently_used_file_handler.cxx
+++ shell/source/unix/sysshell/recently_used_file_handler.cxx
@@ -69,6 +69,7 @@
 #include <vector>
 #include <algorithm>
 #include <functional>
+#include <string.h>
 
 namespace /* private */ {
     
--- svx/source/msfilter/mscodec.cxx
+++ svx/source/msfilter/mscodec.cxx
@@ -42,6 +42,7 @@
 
 #include <osl/diagnose.h>
 #include <algorithm>
+#include <string.h>
 #include <tools/solar.h>
 
 namespace svx {
--- tools/inc/bootstrp/command.hxx
+++ tools/inc/bootstrp/command.hxx
@@ -36,7 +36,7 @@
 #ifndef COMMAND_HXX
 #define COMMAND_HXX
 
-#include <iostream.h>
+#include <iostream>
 
 #include <tools/stream.hxx>
 #define STRLEN 100
--- tools/source/fsys/urlobj.cxx
+++ tools/source/fsys/urlobj.cxx
@@ -81,6 +81,8 @@
 #define INCLUDED_LIMITS
 #endif
 
+#include <string.h>
+
 namespace unnamed_tools_urlobj {} using namespace unnamed_tools_urlobj;
 	// unnamed namespaces don't work well yet...
 
--- unoxml/source/dom/attr.cxx
+++ unoxml/source/dom/attr.cxx
@@ -36,6 +36,7 @@
 #include "attr.hxx"
 #include "element.hxx"
 #include <com/sun/star/xml/dom/DOMException.hdl>
+#include <string.h>
 
 namespace DOM
 {
--- unoxml/source/dom/attributesmap.cxx
+++ unoxml/source/dom/attributesmap.cxx
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "attributesmap.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- unoxml/source/dom/characterdata.cxx
+++ unoxml/source/dom/characterdata.cxx
@@ -36,6 +36,7 @@
 #include <com/sun/star/xml/dom/events/XDocumentEvent.hpp>
 #include "characterdata.hxx"
 #include "../events/mutationevent.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- unoxml/source/dom/document.cxx
+++ unoxml/source/dom/document.cxx
@@ -54,6 +54,8 @@
 #include "../events/uievent.hxx"
 #include "../events/mouseevent.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
--- unoxml/source/dom/documenttype.cxx
+++ unoxml/source/dom/documenttype.cxx
@@ -37,6 +37,7 @@
 #include "entitiesmap.hxx"
 #include "notationsmap.hxx"
 
+#include <string.h>
 
 namespace DOM
 {
--- unoxml/source/dom/element.cxx
+++ unoxml/source/dom/element.cxx
@@ -40,6 +40,8 @@
 #include "attributesmap.hxx"
 #include "../events/mutationevent.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
--- unoxml/source/dom/elementlist.cxx
+++ unoxml/source/dom/elementlist.cxx
@@ -35,6 +35,8 @@
 
 #include "elementlist.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
--- unoxml/source/dom/entity.cxx
+++ unoxml/source/dom/entity.cxx
@@ -35,6 +35,8 @@
 
 #include "entity.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
--- unoxml/source/dom/entityreference.cxx
+++ unoxml/source/dom/entityreference.cxx
@@ -34,6 +34,8 @@
  ************************************************************************/
 
 #include "entityreference.hxx"
+#include <string.h>
+
 namespace DOM
 {
   CEntityReference::CEntityReference(const xmlNodePtr /*aNodePtr*/)
--- unoxml/source/dom/node.cxx
+++ unoxml/source/dom/node.cxx
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include <stdio.h>
+#include <string.h>
 #include "node.hxx"
 #include "element.hxx"
 #include "text.hxx"
--- unoxml/source/dom/notation.cxx
+++ unoxml/source/dom/notation.cxx
@@ -34,6 +34,8 @@
  ************************************************************************/
 
 #include "notation.hxx"
+#include <string.h>
+
 namespace DOM
 {
     CNotation::CNotation(const xmlNotationPtr aNotationPtr)
--- unoxml/source/dom/processinginstruction.cxx
+++ unoxml/source/dom/processinginstruction.cxx
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "processinginstruction.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- unoxml/source/fastsax/fastparser.cxx
+++ unoxml/source/fastsax/fastparser.cxx
@@ -66,6 +66,8 @@
 #include "expat/xmlparse.h"
 #endif
 
+#include <string.h>
+
 using ::rtl::OUString;
 using ::rtl::OString;
 using namespace ::std;
--- vcl/unx/source/app/i18n_im.cxx
+++ vcl/unx/source/app/i18n_im.cxx
@@ -37,6 +37,7 @@
 #include "precompiled_vcl.hxx"
 
 #include <stdio.h>
+#include <string.h>
 #ifdef LINUX
 #  ifndef __USE_XOPEN
 #    define __USE_XOPEN
--- writerfilter/source/dmapper/DomainMapper_Impl.hxx
+++ writerfilter/source/dmapper/DomainMapper_Impl.hxx
@@ -74,6 +74,8 @@
 #endif
 #include <map>
 
+#include <string.h>
+
 namespace com{ namespace sun{ namespace star{
         namespace lang{
             class XMultiServiceFactory;
--- writerfilter/source/odiapi/qname/QName.cxx
+++ writerfilter/source/odiapi/qname/QName.cxx
@@ -44,6 +44,7 @@
 #include <stdexcept>
 #include <string>
 #include <stdio.h>
+#include <string.h>
 
 namespace writerfilter
 {
--- writerfilter/unocomponent/debugservices/rtfsl/RTFSLParser.cxx
+++ writerfilter/unocomponent/debugservices/rtfsl/RTFSLParser.cxx
@@ -39,6 +39,7 @@
 
 #include "RTFSLParser.hxx"
 #include <stdio.h>
+#include <string.h>
 #include <wchar.h>
 #include <rtftok/RTFScanner.hxx>
 #include <rtftok/RTFScannerHandler.hxx>
--- writerfilter/unocomponent/debugservices/rtftok/ScannerTestService.cxx
+++ writerfilter/unocomponent/debugservices/rtftok/ScannerTestService.cxx
@@ -39,6 +39,7 @@
 
 #include "ScannerTestService.hxx"
 #include <stdio.h>
+#include <string.h>
 #include <wchar.h>
 #include <rtftok/RTFScanner.hxx>
 #include <rtftok/RTFScannerHandler.hxx>
--- writerfilter/unocomponent/debugservices/rtftok/XMLScanner.cxx
+++ writerfilter/unocomponent/debugservices/rtftok/XMLScanner.cxx
@@ -39,6 +39,7 @@
 
 #include "XMLScanner.hxx"
 #include <stdio.h>
+#include <string.h>
 #include <wchar.h>
 #include <rtftok/RTFScanner.hxx>
 #include <rtftok/RTFScannerHandler.hxx>
--- writerperfect/source/filter/DocumentCollector.hxx
+++ writerperfect/source/filter/DocumentCollector.hxx
@@ -42,6 +42,7 @@
 #include <vector>
 #include <map>
 #include <stack>
+#include <string.h>
 
 class DocumentElement;
 class DocumentHandler;
--- writerperfect/source/filter/TableStyle.cxx
+++ writerperfect/source/filter/TableStyle.cxx
@@ -27,6 +27,7 @@
  * Corel Corporation or Corel Corporation Limited."
  */
 #include <math.h>
+#include <string.h>
 #include "FilterInternal.hxx"
 #include "TableStyle.hxx"
 #include "DocumentElement.hxx"
--- writerperfect/source/filter/TextRunStyle.cxx
+++ writerperfect/source/filter/TextRunStyle.cxx
@@ -36,6 +36,8 @@
 #include <minmax.h>
 #endif
 
+#include <string.h>
+
 ParagraphStyle::ParagraphStyle(WPXPropertyList *pPropList, const WPXPropertyListVector &xTabStops, const WPXString &sName) :
 	mpPropList(pPropList),
 	mxTabStops(xTabStops),
--- xmlhelp/source/com/sun/star/help/HelpCompiler.cxx
+++ xmlhelp/source/com/sun/star/help/HelpCompiler.cxx
@@ -37,6 +37,7 @@
 #include "HelpCompiler.hxx"
 #include <limits.h>
 #include <stdlib.h>
+#include <string.h>
 #include <libxslt/xslt.h>
 #include <libxslt/xsltInternals.h>
 #include <libxslt/transform.h>
--- xmlhelp/source/com/sun/star/help/HelpLinker.cxx
+++ xmlhelp/source/com/sun/star/help/HelpLinker.cxx
@@ -37,6 +37,9 @@
 
 #include <map>
 
+#include <string.h>
+#include <limits.h>
+
 #include <boost/shared_ptr.hpp>
 #include <boost/tokenizer.hpp>
 
--- xmlhelp/source/cxxhelp/provider/databases.cxx
+++ xmlhelp/source/cxxhelp/provider/databases.cxx
@@ -71,6 +71,7 @@
 #endif
 #include "inputstream.hxx"
 #include <algorithm>
+#include <string.h>
 
 // Extensible help
 #include "com/sun/star/deployment/thePackageManagerFactory.hpp"
--- xmloff/inc/xmloff/PropertySetInfoHash.hxx
+++ xmloff/inc/xmloff/PropertySetInfoHash.hxx
@@ -39,6 +39,7 @@
 #include <xmloff/PropertySetInfoKey.hxx>
 #endif
 
+#include <string.h>
 #include <memory>
 
 struct PropertySetInfoHash
--- xmloff/inc/xmloff/nmspmap.hxx
+++ xmloff/inc/xmloff/nmspmap.hxx
@@ -61,6 +61,8 @@
 #include <cppuhelper/weak.hxx>
 #endif
 
+#include <limits.h>
+
 const sal_uInt16 XML_NAMESPACE_XMLNS  = (USHRT_MAX-2);
 const sal_uInt16 XML_NAMESPACE_NONE  = (USHRT_MAX-1);
 const sal_uInt16 XML_NAMESPACE_UNKNOWN  = (USHRT_MAX);
--- xmloff/source/draw/EnhancedCustomShapeToken.cxx
+++ xmloff/source/draw/EnhancedCustomShapeToken.cxx
@@ -43,6 +43,7 @@
 #include <osl/mutex.hxx>
 #endif
 #include <hash_map>
+#include <string.h>
 
 namespace xmloff { namespace EnhancedCustomShapeToken {
 
--- xmlsecurity/source/helper/xsecparser.cxx
+++ xmlsecurity/source/helper/xsecparser.cxx
@@ -43,6 +43,8 @@
 #endif
 #include "cppuhelper/exc_hlp.hxx"
 
+#include <string.h>
+
 namespace cssu = com::sun::star::uno;
 namespace cssxs = com::sun::star::xml::sax;
 
--- binfilter/bf_basic/source/comp/codegen.cxx.old	2007-03-15 16:07:02.000000000 +0100
+++ binfilter/bf_basic/source/comp/codegen.cxx	2007-11-16 19:19:12.000000000 +0100
@@ -531,15 +531,8 @@
 	m_nCnvtdSize = static_cast<S>( aTrnsfrmer.buffer().GetSize() );
 }
 
-void NeverRunsEver()
-{
-    // force instatiation of templates... I dunno why, but I have to do
-	// this to force instatiation of the template. Otherwise using the template
-	// in another code module results in link errors :-(
-	PCodeBuffConvertor< UINT16, UINT32 > aInst1(0,0);	
-	aInst1.convert();
-	PCodeBuffConvertor< UINT32, UINT16 > aInst2(0,0);	
-	aInst2.convert();
-}
+// instantiate for types needed in SbiImage::Load and SbiImage::Save
+template class PCodeBuffConvertor<UINT16, UINT32 >;
+template class PCodeBuffConvertor<UINT32, UINT16>;
 
 }
--- binfilter/inc/bf_xmloff/nmspmap.hxx.old	2007-01-02 19:52:42.000000000 +0100
+++ binfilter/inc/bf_xmloff/nmspmap.hxx	2007-11-16 19:28:59.000000000 +0100
@@ -36,6 +36,8 @@
 #ifndef _XMLOFF_NMSPMAP_HXX
 #define _XMLOFF_NMSPMAP_HXX
 
+#include <limits.h>
+
 #ifndef __SGI_STL_HASH_MAP
 #include <hash_map>
 #endif
--- binfilter/inc/bf_xmloff/PropertySetInfoHash.hxx.old	2005-09-09 19:12:49.000000000 +0200
+++ binfilter/inc/bf_xmloff/PropertySetInfoHash.hxx	2007-11-16 19:26:51.000000000 +0100
@@ -35,6 +35,8 @@
 #ifndef _XMLOFF_PROPERTYSETINFOHASH_HXX
 #define _XMLOFF_PROPERTYSETINFOHASH_HXX
 
+#include <string.h>
+
 #ifndef _XMLOFF_PROPERTYSETINFOKEY_HXX
 #include <bf_xmloff/PropertySetInfoKey.hxx>
 #endif
Index: bf_forms/source/component/forms_ComboBox.cxx
===================================================================
RCS file: /cvs/framework/binfilter/bf_forms/source/component/forms_ComboBox.cxx,v
retrieving revision 1.7
diff -u -u -r1.7 forms_ComboBox.cxx
--- binfilter/bf_forms/source/component/forms_ComboBox.cxx	27 Oct 2006 13:59:08 -0000	1.7
+++ binfilter/bf_forms/source/component/forms_ComboBox.cxx	25 Jan 2008 16:29:51 -0000
@@ -97,6 +97,8 @@
 #include <com/sun/star/sdbcx/XColumnsSupplier.hpp>
 #endif
 
+#include <limits.h>
+
 namespace binfilter {
 
 	using namespace ::dbtools;
--- bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
+++ bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
@@ -46,6 +46,7 @@
 #include <typelib/typedescription.hxx>
 #include <uno/any2.h>
 
+#include <string.h>
 #include "share.hxx"
 
 
